SAMLPortal/Models/GlobalSettings.cs

```diff
@@ -20,29 +22,53 @@ public static class GlobalSettings
 		/// </summary>
 		public static void InitSettingsFromEnvironment()
 		{
-			DotNetEnv.Env.Load("./.env");
-			IDictionary environment = Environment.GetEnvironmentVariables();
+			FillSettingsArray();
 
-			if (Environment.GetEnvironmentVariable("SP_FORCE_ENV_REPOPUL") == "1")
+			string configurationPath = GlobalSettings.Get("CONFIG_PATH");
+			if (configurationPath != null)
 			{
-				// Re-fill databse with environment configuration
-				foreach (var key in environment.Keys)
+				if (Directory.Exists(configurationPath))
 				{
-					if (key.ToString().StartsWith("SP_") && !key.ToString().StartsWith("SP_MYSQL"))
+					// Ensure trailing slash is present
+					configurationPath = configurationPath.TrimEnd('/') + '/';
+
+					var configPath = configurationPath + "config.env";
+					if (!File.Exists(configPath))
 					{
-						string shortenedKey = key.ToString().Remove(0, 3);
-						_appSettings.Add(shortenedKey, environment[key].ToString());
+						using (FileStream fs = File.Create(configPath)) { }
+						Helpers.WriteEnvVariableToFile(configPath, "SP_CONFIG_SETUPASSISTANT_STEP", "0");
 					}
-				}
 
-				UpdateDatabaseSettings();
+					DotNetEnv.Env.Load(configPath);
+					FillSettingsArray();
+				}
+				else
+				{
+					throw new Exception("The directory " + configurationPath + " does not exists. Please create it first.");
+				}
 			}
 			else
 			{
-				// Store database configuration in global settings
-				UpdateFromDatabase();
+				throw new Exception("Environment variable SP_CONFIG_PATH undefined. Please set it to a correct directory.");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/manual/zegorax/SAMLPortal/pullRequest?prid=5343548)